tests/test_other.py

```diff
@@ -1483,6 +1483,29 @@ def test_include_file(self, args):
     result = self.run_js('a.out.js', engine=config.NODE_JS)
     self.assertContained('|hello from a file wi|', result)
 
+  @parameterized({
+    'default': ([],),
+    'proxy-main': (['-sPROXY_TO_PTHREAD', '-sEXIT_RUNTIME'],),
+  })
+  @node_pthreads
+  def test_preload_with_pthread(self, args):
```

Comment:
Yes - if I only change file_packager.py, but don't change js_manipulation.py, then the test which proxies to pthread fails with error
```
worker.js onmessage() captured an uncaught exception: TypeError: Cannot read property 'slice' of undefined
pthread sent an error! undefined:undefined: Cannot read property 'slice' of undefined

/private/var/folders/70/j6dv6jq15ml8729ps84cctl00000gp/T/tmppl3nk87t/emscripten_test_other_x02xzu0e/a.out.js:360
      throw ex;
```

The default test is to ensure that changes to `file_packager.py` is not breaking preloading with pthread enabled

Reply:
Sorry, I mean if you revert everything except this test.. does it pass?   Another way of putting it:  Does this test fail without the fixes in this PR?   If not, can we make it so that it does?

Reply:
We would need to be able to detect the fact that the files were re-loaded when a thread was first created I guess?   How did you observe this bug originally?  i.e. what was the side effect?

Reply:
> Does this test fail without the fixes in this PR? If not, can we make it so that it does?

I couldn't find the right way to do this -- I observed the failure when our .data file is not hosted in the server root (and so is provided via overloading getPreloadedPackage). In this case, making a request for `/MyLibrary.data` fails. Can you point me to how we can make it so that the .data file is not available by default in server root and requires use of `getPreloadedPackage`? 

Overall, you are right that the tests don't explicitly check what was failing, but rather are ensuring that early return isn't breaking file preloading.

Reply:
nvm, found the test in test_browser.py; will add another one

Reply:
Should I remove the `test_preload_with_pthread` tests, or are we okay to keep both the browser test and this one @sbc100 ?